pkg/controller/framework/informers/factory.go

```diff
@@ -27,6 +28,56 @@ import (
 	"k8s.io/kubernetes/pkg/watch"
 )
 
+type SharedInformerFactory interface {
+	Start(time.Duration)
+
+	Pod() framework.SharedIndexInformer
+	Node() framework.SharedIndexInformer
+	Namespace() framework.SharedIndexInformer
+	PersistentVolumeClaim() framework.SharedIndexInformer
+	PersistentVolume() framework.SharedIndexInformer
+}
+
+type sharedInformerFactory struct {
+	client    clientset.Interface
+	informers map[reflect.Type]framework.SharedIndexInformer
+}
+
+func NewSharedInformerFactory(client clientset.Interface) SharedInformerFactory {
+	return &sharedInformerFactory{
+		client:    client,
+		informers: make(map[reflect.Type]framework.SharedIndexInformer),
+	}
+}
+
+func (s *sharedInformerFactory) Start(resyncPeriod time.Duration) {
+	s.informers[reflect.TypeOf(&api.Pod{})] = CreateSharedPodIndexInformer(s.client, resyncPeriod)
+	s.informers[reflect.TypeOf(&api.Node{})] = CreateSharedNodeIndexInformer(s.client, resyncPeriod)
+	s.informers[reflect.TypeOf(&api.Namespace{})] = CreateSharedNamespaceIndexInformer(s.client, resyncPeriod)
+	s.informers[reflect.TypeOf(&api.PersistentVolumeClaim{})] = CreateSharedPVCIndexInformer(s.client, resyncPeriod)
+	s.informers[reflect.TypeOf(&api.PersistentVolume{})] = CreateSharedPVIndexInformer(s.client, resyncPeriod)
+}
+
+func (f *sharedInformerFactory) Pod() framework.SharedIndexInformer {
+	return f.informers[reflect.TypeOf(&api.Pod{})]
```

Comment:
You should be calling this method before you've called `.Start` (so you can register ahead of time), which means this would most often return `nil`.

If you look at the downsteam implementation, you'll notice that we're lazily initializing these so that we only create watches for things that are requested and it also ensures that we never get nil.